src/Draco.Compiler/Internal/Binding/Binder_UntypedLvalue.cs

```diff
@@ -54,4 +98,58 @@ private UntypedLvalue BindIllegalLvalue(SyntaxNode syntax, ConstraintSolver cons
             location: syntax?.Location));
         return new UntypedIllegalLvalue(syntax);
     }
+
+    private UntypedLvalue BindIndexLvalue(IndexExpressionSyntax index, ConstraintSolver constraints, DiagnosticBag diagnostics)
+    {
+        var receiver = this.BindExpression(index.Indexed, constraints, diagnostics);
+        if (receiver is UntypedReferenceErrorExpression err)
+        {
+            return new UntypedIllegalLvalue(index);
+        }
+        var args = index.IndexList.Values.Select(x => this.BindExpression(x, constraints, diagnostics)).ToImmutableArray();
+        var returnType = constraints.AllocateTypeVariable();
+        var promise = constraints.Substituted(receiver.TypeRequired, () =>
+        {
+            var indexers = constraints.Unwrap(receiver.TypeRequired).Members.OfType<PropertySymbol>().Where(x => x.IsIndexer).Select(x => x.Setter).OfType<FunctionSymbol>().ToImmutableArray();
+            if (indexers.Length == 0)
+            {
+                diagnostics.Add(Diagnostic.Create(
+                    template: SymbolResolutionErrors.NoSettableIndexerInType,
+                    location: index.Location,
+                    receiver.ToString()));
+                constraints.Unify(returnType, new ErrorTypeSymbol("<error>"));
+                return ConstraintPromise.FromResult<FunctionSymbol>(new NoOverloadFunctionSymbol(args.Length + 1));
+            }
+            var overloaded = constraints.Overload(indexers, args.Select(x => x.TypeRequired).Append(returnType).ToImmutableArray(), out var gotReturnType);
+            constraints.Unify(returnType, gotReturnType);
+            return overloaded;
+        });
+        promise.ConfigureDiagnostic(diag => diag
+            .WithLocation(index.Location));
+
+        return new UntypedIndexSetLvalue(index, promise, receiver, args, returnType);
+    }
+
+    private UntypedLvalue SymbolToLvalue(SyntaxNode syntax, Symbol symbol, ConstraintSolver constraints, DiagnosticBag diagnostics)
+    {
+        switch (symbol)
+        {
+        case FieldSymbol field:
+            return new UntypedFieldLvalue(syntax, null, field);
+        case PropertySymbol prop:
+            var setter = prop.Setter;
+            if (setter is null)
+            {
+                diagnostics.Add(Diagnostic.Create(
+                    template: SymbolResolutionErrors.CannotSetGetOnlyProperty,
+                    location: syntax?.Location,
+                    prop.FullName));
+                setter = new NoOverloadFunctionSymbol(1);
+            }
+            return new UntypedPropertySetLvalue(syntax, setter, null);
```

Comment:
While this is abstract syntax, please order so it still corresponds to the concrete syntactical elements. This means, make receiver appear before the setter itself, since the syntax is `receiver.Setter`.
Please fix this for every newly added node where this is flipped